tower-request-modifier/src/lib.rs

```diff
@@ -0,0 +1,173 @@
+extern crate futures;
+extern crate http;
+extern crate tower_service;
+
+use futures::Poll;
+use http::header::{HeaderName, HeaderValue};
+use http::{Request, HttpTryFrom};
+use http::uri::{self, Authority, Scheme, Uri};
+use tower_service::Service;
+
+#[derive(Debug, Clone)]
+pub enum Modification {
```

Comment:
I would try to keep this enum private. Doing so would allow us to add modifications without a breaking change.